src/core/index.js

```diff
@@ -46,153 +46,27 @@ export default function SwaggerUI(opts) {
-      ApisPreset
-    ],
-
-    // Plugins; ( loaded after presets )
-    plugins: [
-    ],
-
-    pluginsOptions: {
-      // Behavior during plugin registration. Can be :
-      // - legacy (default) : the current behavior for backward compatibility – last plugin takes precedence over the others
-      // - chain : chain wrapComponents when targeting the same core component
-      pluginLoadType: "legacy"
-    },
-
-    // Initial state
-    initialState: { },
-
-    // Inline Plugin
-    fn: { },
-    components: { },
-
-    syntaxHighlight: {
-      activated: true,
-      theme: "agate"
-    }
-  }
-
-  let queryConfig = opts.queryConfigEnabled ? parseSearch() : {}
-
   const domNode = opts.domNode
   delete opts.domNode
 
-  const constructorConfig = deepExtend({}, defaults, opts, queryConfig)
-
-  const storeConfigs = {
-    system: {
-      configs: constructorConfig.configs
-    },
-    plugins: constructorConfig.presets,
-    pluginsOptions: constructorConfig.pluginsOptions,
-    state: deepExtend({
-      layout: {
-        layout: constructorConfig.layout,
-        filter: constructorConfig.filter
-      },
-      spec: {
-        spec: "",
-        // support Relative References
-        url: constructorConfig.url,
-      },
-      requestSnippets: constructorConfig.requestSnippets
-    }, constructorConfig.initialState)
-  }
-
-  if(constructorConfig.initialState) {
-    // if the user sets a key as `undefined`, that signals to us that we
-    // should delete the key entirely.
-    // known usage: Swagger-Editor validate plugin tests
-    for (var key in constructorConfig.initialState) {
-      if(
-        Object.prototype.hasOwnProperty.call(constructorConfig.initialState, key)
-        && constructorConfig.initialState[key] === undefined
-      ) {
-        delete storeConfigs.state[key]
-      }
-    }
-  }
+  const { queryConfig, combinedConfig, storeConfig } = getConfigs(opts)
 
-  let inlinePlugin = ()=> {
+  const inlinePlugin = ()=> {
     return {
-      fn: constructorConfig.fn,
-      components: constructorConfig.components,
-      state: constructorConfig.state,
+      fn: combinedConfig.fn,
+      components: combinedConfig.components,
+      state: combinedConfig.state,
     }
   }
 
-  var store = new System(storeConfigs)
-  store.register([constructorConfig.plugins, inlinePlugin])
+  const store = new System(storeConfig)
+  store.register([combinedConfig.plugins, inlinePlugin])
 
-  var system = store.getSystem()
+  const system = store.getSystem()
 
   const downloadSpec = (fetchedConfig) => {
-    let localConfig = system.specSelectors.getLocalConfig ? system.specSelectors.getLocalConfig() : {}
-    let mergedConfig = deepExtend({}, localConfig, constructorConfig, fetchedConfig || {}, queryConfig)
+    const localConfig = system.specSelectors.getLocalConfig ? system.specSelectors.getLocalConfig() : {}
+    const mergedConfig = deepExtend({}, localConfig, combinedConfig, fetchedConfig || {}, queryConfig)
```

Comment:
Additionally: doesn't this order mean that we will always overwrite values in `localConfig` with the default config values which are merged into `combinedConfig`?

We also can't specify `configUrl` inside the local config as well.

Reply:
Addressed by dissecting and composing the config again together in https://github.com/swagger-api/swagger-ui/pull/9815